src/http/server.rs

```diff
@@ -34,7 +48,10 @@ async fn graphql_request(req: Request<Body>, server_ctx: &ServerContext) -> Resu
     response = response.set_cache_control(ttl);
   }
 
-  response.to_response()
+  let resp = response.to_response().unwrap_or_default();
+  let headers = server_ctx.server.get_response_headers();
+
+  add_headers(resp, headers)
```

Comment:
I think we should add the headers before we convert the response to a graphql response. The flow should remain the same after `response.to_response()`

Reply:
Added the set_response_header function to GraphQL response for the same.